### PR TITLE
Address clippy lints

### PR DESCRIPTION
**Pull request by erickt**

### CODE REVIEW

File: tuf/src/metadata.rs [hunk 1]
```diff
@@ -3036,7 +3036,7 @@ mod test {
     //                              (,,(,    \.         `.   ____,-`.,
     //                           (,'     `/   \.   ,--.___`.'
     //                       ,  ,'  ,--.  `,   \.;'         `
-    //                        `{o, {    \  :    \;
+    //                        `(o, /    \  :    \;
```
Comment (kellerb) [inline tuf/src/metadata.rs:3039]: did not expect clippy to have preferences on the depictions of wyrms

File: tuf/src/repository/track_repo.rs [hunk 2]
```diff
@@ -117,8 +117,7 @@ where
             let mut buf = Vec::new();
             metadata.read_to_end(&mut buf).await?;
 
-            let () = self
```
Comment (kellerb) [inline tuf/src/repository/track_repo.rs]: i actually like these unit bindings, makes it very obvious to the reader nothing important is being ignored (i know `must_use` is for that, but sometimes there are exceptions

not that you shouldn't get rid of it here, just be prepared for push back in //src/sys/pkg :)

**Review by kellerb — APPROVED** [review]

File: tuf/src/metadata.rs (hunk 1 above)
Comment (erickt) [reply tuf/src/metadata.rs:3039]: This wasn't *really* a clippy lint, but those `{` always screwed up bracket matching in vim and vscode, so I decided to finally get rid of it.

File: tuf/src/repository/track_repo.rs (hunk 2 above)
Comment (erickt) [reply tuf/src/repository/track_repo.rs]: Oh I generally agree, just in this case the track_repo is a test helper crate, so I wasn't too worried about the return type changing. 

But this seems like a worthwhile lint to disable in general, so I added it.

**Pull request merged**